src/sonic_ax_impl/mibs/__init__.py

```diff
@@ -504,7 +510,20 @@ def get_non_host_dbs(dbs):
             return dbs
         else:
             return dbs[1:]
-        
+
+    @staticmethod
+    def get_start_idx_of_non_host(dbs):
+        """
+        From the list of all dbs, return the list of dbs
+        which will have interface related tables.
+        For single namespace db, return the single db.
+        For multiple namespace dbs, return all dbs except the
+        host namespace db which is the first db in the list.
+        """
+        if len(dbs) == 1:
+            return 0
+        else:
+            return 1
 
     @staticmethod
     def init_namespace_sync_d_interface_tables(dbs):
```

Comment:
>init_namespace_sync_d [](start = 8, length = 21)

I see a common pattern in all the functions like `init_namespace_sync_d*`. They just call the single db function and merge all returned items respectively.

How about abstract this high level function and return a dictionary of list, in  this case the type is `db -> { if_name_map, if_alias_map, if_id_map, oid_sai_map, oid_name_map, if_oid_namespace }`

Reply:
In the MIB implementations, we generally use a single dict to look up the if_index etc. If we have a dict of lists, then we will have to loop through this dict for each instance of the db and look for the if_index, which will make the code look bulky in the places where these functions are used.